pallets/attestation/src/benchmarking.rs

```diff
@@ -67,12 +82,35 @@ benchmarks! {
 			attester: delegate_acc.into(),
 			delegation_id: Some(delegation_id),
 			revoked: true,
+			deposit: kilt_support::deposit::Deposit {
+				owner: delegate_public.into(),
+				amount: T::Deposit::get(),
+			}
 		}));
 	}
+
+	remove {
+		let d in 1 .. T::MaxParentChecks::get();
+
+		let claim_hash: T::Hash = T::Hashing::hash(b"claim");
+		let ctype_hash: T::Hash = T::Hash::default();
+
+		let (root_public, _, delegate_public, delegation_id) = setup_delegations::<T>(d, ONE_CHILD_PER_LEVEL.expect(">0"), Permissions::ATTEST | Permissions::DELEGATE)?;
+		let root_acc: T::AccountId = root_public.into();
+		let delegate_acc: T::AccountId = delegate_public.into();
+		T::Currency::make_free_balance_be(&delegate_acc, T::Deposit::get() + T::Deposit::get());
+
+		// attest with leaf account
+		Pallet::<T>::add(RawOrigin::Signed(delegate_acc).into(), claim_hash, ctype_hash, Some(delegation_id))?;
+		// revoke with root account, s.t. delegation tree needs to be traversed
+	}: _(RawOrigin::Signed(root_acc.clone()), claim_hash, d)
+	verify {
+		assert!(!Attestations::<T>::contains_key(claim_hash));
```

Comment:
C: I would rather move (or add) the deposit checks to the test cases. Plus, I would check for balance state after deletion, as I mentioned in another comment.

Reply:
1e99025